pkg/cloud/cloud_test.go

```diff
@@ -2059,6 +2059,25 @@ func TestCreateSnapshot(t *testing.T) {
 			},
 			expErr: nil,
 		},
+		{
```

Comment:
Once you add in the suggested `isARN` validation, consider adding an extra unit test for it.

Otherwise lgtm, thanks!

Reply:
sounds good thank you !